generator/tree.go

```diff
@@ -363,11 +366,14 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
 				if subtype, ok := combinedTypes[index.Type]; ok {
 					if prevType == subtype {
 						metric.Indexes = metric.Indexes[:len(metric.Indexes)-1]
+					if prev2Type == subtype {
```

Comment:
I think you wanted an else if here?

```suggestion
					} else if prev2Type == subtype {
```